client/js/socket-events/uploads.js

```diff
@@ -1,10 +1,22 @@
 "use strict";
 
 const socket = require("../socket");
-const wrapCursor = require("undate").wrapCursor;
+const updateCursor = require("undate").update;
 
 socket.on("upload:success", (url) => {
-	const fullURL = new URL(url, location);
+	const fullURL = (new URL(url, location)).toString();
 	const textbox = document.getElementById("input");
-	wrapCursor(textbox, fullURL, " ");
+	const initStart = textbox.selectionStart;
+
+	// Get the text before the cursor, and append a space if it's not in the beginning
+	const headToCursor = textbox.value.substr(0, textbox.selectionStart) + (textbox.selectionStart > 0 ? " " : "");
+
+	// Get the remaining text after the cursor and append a space
```

Comment:
prepend?

Reply:
it appends in relation the text before cursor, not the link